drivers/at86rf215/Kconfig

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+menuconfig KCONFIG_MODULE_AT86RF215
+    bool "Configure AT86RF215 driver"
+    depends on MODULE_AT86RF215
+    help
+        Configure the AT86RF215 driver using Kconfig.
+
+if KCONFIG_MODULE_AT86RF215
+
+config AT86RF215_USE_CLOCK_OUTPUT
+    bool "Enable/Disable Onchip Clock Source"
+    default n
+    help
+        Enable this to set the clock output of the AT86RF215
+        as the clock source on the board.
+        By Default it is turned off to save energy.
+
+config AT86RF215_TRIM_VAL
```

Comment:
This will always apply a trim when using Kconfig, as it will always be defined. Maybe add a switch to turn trimming ON and OFF?

Reply:
Trimming seems to be required for the operation from internal clock so as to set the proper frequecny. Trimming would only be applied provided  CONFIG_AT86RF215_USE_CLOCK_OUTPUT = y in Kconfig

Reply:
Hmm I think the output is mentioned on the documentation of the trim value so one can measure the internal oscillator. Once it is calibrated the output can be turned off. At least that's what I understood. @benpicco?

Reply:
Trimming is not required, it can be used to enhance the radio performance by making sure the center frequency it sends at matches the expected reference.

The trim value is board specific, depending on the characteristics of the capacitors used and might even differ between boards with the same hardware. (Although in practice I didn't see that with boards from the same production batch)

The clock will work just fine if no trim is applied.

Reply:
This is still not resolved. The trim value will always get the value 0 when using Kconfig and will be applied. We need to enable the trimming with another symbol.

Reply:
I dont quite understand. We can always remove the default.

Reply:
If you remove the default you end up with a defined macro with no value

Reply:
I mean having something like:
```
config AT86RF215_TRIM_VAL_EN
    bool "Enable crystal oscillator trimming"

config AT86RF215_TRIM_VAL
    int "Trim value for the crystal oscillator"
    range 0 15
    default 0
    depends on AT86RF215_TRIM_VAL_EN
```

Reply:
Thanks for the suggestion. It has been updated.